### DISCUSSION TITLE
Is it possible to view custom events without defining them as a 'goal'?

### DISCUSSION THREAD

**Discussion by jerome-toole** [plausible/analytics#1772] [Anything Else]
I have a few custom events that are fired on delays or failures to a user request. They aren't goals but currently the only way I can see of viewing those events is by defining them as a goal. At the moment, this is mostly a semantic issue but if Plausible ever had an overall 'Conversion Rate' stat, then I wouldn't want these events to be included in that as they are in fact errors.

In addition to this, it would be great to be able to view events that haven't been registered in any way in the Dashboard. Just send them from the website and see them appear somewhere.

**Comment by ak-beyond** [comment]
A couple follow-on questions around custom events. 
1) Will custom events still be recorded in the events API despite not being set up as goals? I'm thinking about retrieving these stats programmatically and displaying within a CMS.
2) When managing multiple sites in the dashboard, would it be necessary to create a goal for each site instance that I manage, or is there the capability to create a global goal in one place that is shared across all sites?

**Reply by ukutaht** [reply]
1. Yes, custom events are recorded regardless of whether a goal is set up.
2. There's no way to bulk-create goals so yes you would have to set them up on each site separately

**Comment by fxmb** [comment]
To follow up here: (When) can we design multiple conversion flows in seperate entities? Just one goal funnel is quickly not enough anymore and makes plausible really unattractive beyond a very small business stage...

**Reply by metmarkosaric** [reply]
thanks for the feedback @fxmb! could you share a bit more details please on what exactly you have in mind please? thanks!

**Reply by fxmb** [reply]
Well... imagine I have a classic E-Commerce website so my goal could be the basic conversion funnel of such a site
1. Users on Site
2. Users add items to Cart
3. Users go to checkout
4. users actually buy
you get the idea.

But: What if I had a second conversion funnel, hence "Goal". E.g. my ecommerce shop has a blog section and I want to track how many people go on a specifc blog page, do some sort of action and then go back to the main ecommerce section via a CTA. This would be a second conversion funnel. 

Currently I see no option how to model both of these in different funnels yet on the same dashboard. This is an absolute basic feature imo.

**Reply by metmarkosaric** [reply]
ah see! thanks for sharing the details! it's something we can consider when we build #171

**Comment by quantizor** [comment]
This has been solved by the community. The recommended approach today is [`postcss-styled-syntax`](https://github.com/hudochenkov/postcss-styled-syntax) — it's actively maintained (v0.7.1, Jan 2025) and works with Stylelint v14+ as a custom syntax for styled-components, Emotion, and other tagged template literal CSS-in-JS libraries.

```json
{
  "overrides": [
    {
      "files": ["**/*.{js,jsx,ts,tsx}"],
      "customSyntax": "postcss-styled-syntax"
    }
  ]
}
```

For context: Stylelint removed its built-in CSS-in-JS support after v14 in favor of pluggable custom syntaxes, and the old `stylelint-processor-styled-components` (from this org) and `@stylelint/postcss-css-in-js` are both deprecated/archived. `postcss-styled-syntax` is the successor and the only actively maintained option.

The alternative mentioned in the thread (`postcss-styled-components` by 43081j) has not been updated since January 2023.
